api/courses.go

```diff
@@ -363,7 +369,78 @@ func (r coursesRoutes) uploadVOD(c *gin.Context) {
 		return
 	}
 	key := uuid.NewV4().String()
-	err = r.UploadKeyDao.CreateUploadKey(key, stream.ID)
+	err = r.UploadKeyDao.CreateUploadKey(key, stream.ID, model.VideoTypeCombined)
+	if err != nil {
+		_ = c.Error(tools.RequestError{
+			Status:        http.StatusInternalServerError,
+			CustomMessage: "can not create upload key",
+			Err:           err,
+		})
+		return
+	}
+	workers := r.WorkerDao.GetAliveWorkers()
+	if len(workers) == 0 {
+		_ = c.Error(tools.RequestError{
+			Status:        http.StatusInternalServerError,
+			CustomMessage: "no workers available",
+			Err:           err,
+		})
+		return
+	}
+	w := workers[getWorkerWithLeastWorkload(workers)]
+	u, err := url.Parse("http://" + w.Host + ":" + WorkerHTTPPort + "/upload?" + c.Request.URL.Query().Encode() + "&key=" + key)
+	if err != nil {
+		_ = c.Error(tools.RequestError{
+			Status:        http.StatusInternalServerError,
+			CustomMessage: fmt.Sprintf("parse proxy url: %v", err),
+			Err:           err,
+		})
+		return
+	}
+	p := httputil.NewSingleHostReverseProxy(u)
+	p.Director = func(req *http.Request) {
+		req.URL.Scheme = u.Scheme
+		req.URL.Host = u.Host
+		req.Host = u.Host
+		req.URL.Path = u.Path
+		req.URL.RawQuery = u.RawQuery
+	}
+	p.ServeHTTP(c.Writer, c.Request)
+}
+
+func (r coursesRoutes) uploadVODMedia(c *gin.Context) {
```

Comment:
Why do we need another handler? Can you elaborate & add a comment what this is used for?